server/consumer.go

```diff
@@ -1645,18 +1577,17 @@ func (o *consumer) processNextMsgReq(_ *subscription, c *client, _, reply string
 	defer o.mu.Unlock()
 
 	s, mset, js := o.srv, o.mset, o.js
-	if mset == nil || o.sendq == nil {
+	if mset == nil {
 		return
 	}
-	sendq := o.sendq
+	outq := o.outq
 
 	sendErr := func(status int, description string) {
 		// Needs to be unlocked to send err.
 		o.mu.Unlock()
 		defer o.mu.Lock()
 		hdr := []byte(fmt.Sprintf("NATS/1.0 %d %s\r\n\r\n", status, description))
-		pmsg := &jsPubMsg{reply, reply, _EMPTY_, hdr, nil, nil, 0}
-		sendq <- pmsg // Send error message.
+		outq.send(&jsPubMsg{reply, reply, _EMPTY_, hdr, nil, nil, 0, nil})
```

Comment:
Not going to repeat that everywhere, but here again you would not need the "unlock/defer lock" in this function, etc..